webclient/http1/src/main/java/io/helidon/webclient/http1/Http1CallChainBase.java

```diff
@@ -411,23 +411,31 @@ public int read() {
             if (finished) {
                 return -1;
             }
-            ensureBuffer(512);
-            if (finished || currentBuffer == null) {
+            try {
+                ensureBuffer(512);
+                if (finished || currentBuffer == null) {
+                    return -1;
+                }
+                return currentBuffer.read();
+            } catch (DataReader.InsufficientDataAvailableException e) {
```

Comment:
Yes, that's a bit SSE specific (ending is determined by server closing the connection). Having said that, because this is an `InputStream` it should be throwing an `IOException` and our own runtime exception. I'll think about this a bit more.